src/LowMachEquationSystem.C

```diff
@@ -2985,6 +3074,173 @@ ContinuityEquationSystem::register_symmetry_bc(
   }
 }
 
+//--------------------------------------------------------------------------
+//-------- register_abltop_bc ----------------------------------------------
+//--------------------------------------------------------------------------
+void
+ContinuityEquationSystem::register_abltop_bc(
+  stk::mesh::Part *part,
+  const stk::topology &partTopo,
+  const ABLTopBoundaryConditionData &abltopBCData)
+{
+  auto userData = abltopBCData.userData_;
+  if (!userData.ABLTopBC_) {
+    SymmetryBoundaryConditionData symData(abltopBCData.boundaryConditions_);
+    register_symmetry_bc(part, partTopo, symData);
+    return;
+  }
+
+  // algorithm type
+  const AlgorithmType algType = TOP_ABL;
+
+  ScalarFieldType &pressureNone = pressure_->field_of_state(stk::mesh::StateNone);
+  VectorFieldType &dpdxNone = dpdx_->field_of_state(stk::mesh::StateNone);
+
+/*
```

Comment:
Should this throw an error if the user doesn't select edge based scheme?

Reply:
I added some statements here to throw an error for a non edge based scheme.